pepys_import/core/store/data_store.py

```diff
@@ -1008,8 +1014,41 @@ def get_platform(
         :type identifier: String
         :param change_id: ID of the :class:`Change` object
         :type change_id: Integer or UUID
+        :param unknown: Whether to create this as an 'unknown' Platform, which won't ask
+                        the user to resolve details, and will assign Unknown nationality
+                        and Platform Types
+        :type unknown: Boolean
         :return: Created Platform entity
         """
+        if unknown:
+            # The importer has specifically said it wants to get a 'unknown platform' (ie. one with an unknown
+            # nationality and platform type) back from this call.
+            if platform_name is not None:
+                # The importer has provided a name, which we assume means they think this name is unique
+                # (like an IMO ID), so we search for this name and return it if it exists
+                # We'll use the first three characters of the platform name as the identifier - as this will have been used
+                # when the unknown platform was automatically created
+                platform = self.find_platform(
+                    name=platform_name, identifier=platform_name[:3], nationality="Unknown"
```

Comment:
If we have an IMO number, I think this is a legitimate `identifier` - and we shouldn't trim it to 3 chars.  I think we should have the full IMO in name _and_ identifier.  At some point in the future the analyst takes a deeper interest in the platform, and finds the name for the platform with that IMO.  Then they can swap the name value for the real name ("Solent Express"), and we'll have valid values in `name` and `identifier`.

Reply:
Changed now.